cs/src/io/unsafe/MemoryHelper.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.ConstrainedExecution;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Bond.IO.Unsafe
+{
+    internal unsafe class MemoryHelper
+    {
+        [DllImport("msvcrt.dll", EntryPoint = "memcpy", CallingConvention = CallingConvention.Cdecl, CharSet = CharSet.Unicode, SetLastError = false)]
```

Comment:
It seems like there isn't a good option to support `memcpy` in a cross-platform way. It would require building platform specific assemblies.

I would suggest to ditch `memcpy` and use `System.Buffer.MemoryCopy` conditionally on 4.6 and otherwise just the loop in C#. We don't yet build a separate version for .NET 4.6 but we can add that and then the `MemoryCopy` will light up.